modLogger/src/org/aion/log/AionLoggerFactory.java

```diff
@@ -22,191 +22,201 @@
-        }
+        context = (LoggerContext) LoggerFactory.getILoggerFactory();
     }
 
-    public static void init(final Map<String, String> _logModules) {
-        init(_logModules, false, "log");
+    public static void init(Map<String, String> requestedLogLevels) {
+        init(requestedLogLevels, false, "");
     }
 
-    public static void init(final Map<String, String> _logModules, boolean _logToFile, String _logToPath) {
-
-        logModules = _logModules;
-        logFile = _logToFile;
-        logPath = _logToPath;
-
-        loggerContext = (LoggerContext) LoggerFactory.getILoggerFactory();
-
-        /* Toggles file appending configurations */
-        if (logFile) {
-
-            /* Initialize Rolling-File-Appender */
-            String fileName = logPath + "/aionCurrentLog.dat";
-            fileAppender = new RollingFileAppender();
-            fileAppender.setContext(loggerContext);
-            fileAppender.setName("aionlogger");
-            fileAppender.setFile(fileName);
-
-            /* Initialize Triggering-Policy (CONDITION) */
-            SizeBasedTriggeringPolicy tp = new SizeBasedTriggeringPolicy();
-            tp.setContext(loggerContext);
-            tp.start();
-
-            /* Initialize Rolling-Policy (BEHAVIOUR) */
-            SizeAndTimeBasedRollingPolicy rp = new SizeAndTimeBasedRollingPolicy();
-            rp.setContext(loggerContext);
-
-            /*
-             * To modify period of each rollover;
-             * https://logback.qos.ch/manual/appenders.html#TimeBasedRollingPolicy
-             * (Currently set to PER DAY)
-             */
-            FileNamePattern fnp =
-                    new FileNamePattern(
-                            logPath + "/%d{yyyy/MM, aux}/aion.%d{yyyy-MM-dd}.%i.log", loggerContext);
-            rp.setFileNamePattern(fnp.getPattern());
-
-            /*
-             * To modify size of each rollover file;
-             * https://logback.qos.ch/manual/appenders.html#SizeAndTimeBasedRollingPolicy
-             * (Currently set to 100MB)
-             */
-            rp.setMaxFileSize(new FileSize(100 * 1000 * 1000));
-            rp.setParent(fileAppender);
-            rp.start();
-
-            /* Sets TRIGGER & ROLLING policy */
-            fileAppender.setTriggeringPolicy(tp);
-            fileAppender.setRollingPolicy(rp);
-
-            /* Set fileAppender configurations */
-            fileAppender.setContext(loggerContext);
-            fileAppender.setEncoder(encoder);
-            fileAppender.setAppend(true);
-            fileAppender.start();
+    private static Map<LogEnum, Level> constructModuleLoglevelMap(Map<String, String> _moduleToLevelMap) {
+        // condition the input hashmap so keys are all uppercase
+        Map<String, String> moduleToLevelMap = new HashMap<>();
+        _moduleToLevelMap.forEach((module, level) -> {
+            moduleToLevelMap.put(module.toUpperCase(), level);
+        });
+
```

Comment:
can ignore the uppercase transfer if we do it at CfgLog. Means don't need to declare another map.

Reply:
Second, I want this class to be less fragile than before; the principle I'm following is to not depend on any assumptions made in another class or module, unless it provable degrades performance. Since constructModuleLoglevelMap() happens only on init, this small object allocation (weak reference, gets GC'ed fairly quickly) actually makes this class more robust.

Reply:
okay